models/spring-ai-bedrock-converse/src/main/java/org/springframework/ai/bedrock/converse/BedrockProxyChatModel.java

```diff
@@ -195,7 +199,7 @@ public ChatResponse call(Prompt prompt) {
 		if (!this.isProxyToolCalls(prompt, this.defaultOptions) && chatResponse != null
 				&& this.isToolCall(chatResponse, Set.of("tool_use"))) {
 			var toolCallConversation = this.handleToolCalls(prompt, chatResponse);
-			return this.call(new Prompt(toolCallConversation, prompt.getOptions()));
+			return this.internalCall(new Prompt(toolCallConversation, prompt.getOptions()), chatResponse);
```

Comment:
We need test coverage to test this flow - especially to verify the final ChatResponse containing the preceding one.